### PR TITLE
Resizeable window

### PR DESCRIPTION
**Pull request by Sajdkick**
Registered a function callback in MainWindow that triggers when the window is resized. It then changes the size of the buffers and deferredLighting in RenderManager.

### CODE REVIEW

**Comment by Chainsawkitten** [comment]
Why remove jsoncpp?

**Comment by Sajdkick** [comment]
@Chainsawkitten It was picked up by Sourcetree and it wouldn't let me merge. So i removed it from tracking.

**Comment by Chainsawkitten** [comment]
@Sajdkick Well, we need it to compile.

**Comment by Sajdkick** [comment]
@Chainsawkitten I guess i have to add it again then!

File: src/Engine/MainWindow.cpp [hunk 1]
```diff
@@ -1,12 +1,15 @@
 #include "Util/Log.hpp"
 #include "MainWindow.hpp"
+#include "Manager/Managers.hpp"
+#include "Manager/RenderManager.hpp"
 #include <GL/glew.h>
 #include <GLFW/glfw3.h>
 
 MainWindow* MainWindow::instance = nullptr;
+void window_size_callback(GLFWwindow* window, int width, int height);
```
Comment (Chainsawkitten) [inline src/Engine/MainWindow.cpp]: Incorrectly named forward declaration

File: src/Engine/MainWindow.cpp [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
 
 MainWindow::MainWindow(int width, int height, bool fullscreen, bool borderless, const char* title, bool debugContext) {
-    glfwWindowHint(GLFW_RESIZABLE, GL_FALSE);
+    glfwWindowHint(GLFW_RESIZABLE, GL_TRUE);
```
Comment (Chainsawkitten) [inline src/Engine/MainWindow.cpp]: Should be GLFW_TRUE

File: src/Engine/MainWindow.cpp [hunk 3]
```diff
@@ -1,12 +1,15 @@
...
     
     if (borderless)
         glfwWindowHint(GLFW_DECORATED, GL_FALSE);
```
Comment (Chainsawkitten) [inline src/Engine/MainWindow.cpp]: Should be GLFW_FALSE

File: src/Engine/MainWindow.cpp [hunk 4]
```diff
@@ -89,3 +101,11 @@ void MainWindow::SwapBuffers() {
...
     return window;
 }
+
+void WindowSizeCallback(GLFWwindow* window, int width, int height)
+{
+
+    MainWindow::GetInstance()->SetSize(width, height);
+    Managers().renderManager->UpdateBufferSize();
+
+}
```
Comment (Chainsawkitten) [inline src/Engine/MainWindow.cpp]: Missing newline at end of file

File: src/Engine/MainWindow.hpp [hunk 5]
```diff
@@ -45,10 +45,17 @@ class MainWindow {
         
         /// Get the size of the window.
         /**
-         * @return The size of the window in pixels.
-         */
+        * @return The size of the window in pixels.
```
Comment (Chainsawkitten) [inline src/Engine/MainWindow.hpp]: Incorrect doxygen indentation (align * with previous line).

File: src/Engine/MainWindow.cpp (hunk 4 above)
Comment (Sajdkick) [inline src/Engine/MainWindow.cpp]: Fixed

**Pull request merged**